lms/djangoapps/courseware/views.py

```diff
@@ -390,6 +392,12 @@ def _index_bulk_op(request, course_key, chapter, section, position):
     if not registered:
         # TODO (vshnayder): do course instructors need to be registered to see course?
         log.debug(u'User %s tried to view course %s but is not enrolled', user, course.location.to_deprecated_string())
+        if bool(staff_access):
+            usage_key = UsageKey.from_string(course.location.to_deprecated_string()).replace(course_key=course_key)
```

Comment:
Can you use `course_key_from_string_or_404` ( or the one it is being replaced by) method here ?